wetmap/src/components/newModals/setting/index.tsx

```diff
@@ -1,79 +1,47 @@
 
-import SettingsView from './view';
-
+import { signOut } from '../../../supabaseCalls/authenticateSupabaseCalls';
 import { SessionContext } from '../../contexts/sessionContext';
-import { useContext, useState, useEffect } from 'react';
+import { UserProfileContext } from '../../contexts/userProfileContext';
+import SettingsView from './view';
 import React from 'react';
+import { useContext } from 'react';
+import PartnerAccountRequestModal from '../../modals/partnerAccountRequestModal';
+import { ModalContext } from '../../reusables/modal/context';
 
 
 export default function Settings(props) {
   const { activeSession, setActiveSession } = useContext(SessionContext);
-  { /*
-  const [showDangerZone, setShowDangerZone] = useState(false);
-  const [openDialog, setOpenDialog] = useState(false);
-  const [requestCheck, setRequestCheck] = useState([]);
+  const { profile, setProfile } = useContext(UserProfileContext);
   const { modalShow } = useContext(ModalContext);
 
-
-  const checkForRequest = async (id) => {
-    const returnedCheck = await grabRequestById(id);
-    setRequestCheck(returnedCheck);
-  };
-
-  const handlePartnerButton = () => {
-    modalShow(PartnerAccountRequestModal, { keepPreviousModal: true });
-  };
-  useEffect(() => {
-    checkForRequest(activeSession.user.id);
-  }, []);
-
-  const dangerZone = (
-    <div
-      style={{
-        height:       '50px',
-        width:        '90%',
-        color:        'pink',
-        borderRadius: '15px',
-        position:     'absolute',
-        bottom:       '5%',
-        left:         '30%',
-        textAlign:    'center',
-      }}
-    >
-      <div
-        onClick={() => {
-          setOpenDialog(true);
-          setShowDangerZone(false);
-        }}
-        className="AccountDeleteButton"
-      >
-        <Label
-          style={{
-            fontFamily: 'Itim',
-            color:      'maroon',
-            cursor:     'pointer',
-            fontSize:   '1vw',
-          }}
-        >
-          Delete Account
-        </Label>
-      </div>
-    </div>
-  );
-
   const handleLogout = async () => {
     await localStorage.removeItem('token');
     await signOut();
+    props?.onModalCancel?.();
     setActiveSession(null);
-  };  */ }
+  };
 
   const onClose = () => {
     props?.onModalCancel?.();
   };
 
+  let profileType;
+  if (profile.partnerAccount) {
```

Comment:
Typescript is complaining here because profile can be passed to it as a null value and its not allowed to receive a null value. simply add profile && in front of profile.partnerAccount to fix it